src/main/java/com/example/h2/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+package com.example.h2.service.impl;
+
+import com.example.h2.common.PassBasedEnc;
+import com.example.h2.model.User;
+import com.example.h2.pojo.*;
+import com.example.h2.repo.UserRepo;
+import com.example.h2.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Autowired
+    UserRepo repo;
+    @Value("${app.salt}")
+    String salt;
+
+    @Override
+    public ResponseEntity<User> createUser(UserCreateRequest request) {
+        User existingUser = repo.findByEmail(request.getEmail());
+        if (existingUser != null) {
+            return new ResponseEntity<>(HttpStatus.ALREADY_REPORTED);
+        }
+        User newUser = new User();
+        newUser.setEmail(request.getEmail());
+        newUser.setFirstname(request.getFirstname());
+        if (request.getLastname() != null) {
```

Comment:
During create user No need to change null because if we are not setting anyting in first time then by default null value will set.
So in create you can directly set without checking if not null condition

Reply:
okk